src/API/AbstractPluginActions.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+namespace CF\API;
+
+use CF\Integration\DataStoreInterface;
+use CF\Integration\DefaultIntegration;
+use CF\WordPress\Constants\Exceptions\PageRuleLimitException;
+use CF\WordPress\Constants\Exceptions\ZoneSettingFailException;
+use CF\WordPress\Constants\Plans;
+
+abstract class AbstractPluginActions
```

Comment:
We currently assume every plugin to have these 3 endpoints but if they don't have it we should disable the endpoint to reduce the # potential bugs.

1. One solution is to have enable/disable functionality for shared endpoints.   
_Cons: Increases complexity_
2. Other is overriding the function with an empty function.  (Currently this is supported)  
_Cons: Hard to document_

In example if we don't want `login()` we can write:
```
class PluginActions extends AbstractPluginActions
{
   public function login()
   {
   }
}
```

Do you've a better solution? I'm in favor of the second one but I don't feel like it's a good way to do it.